Sources/SPTDataLoaderSwift/DataLoaderWrapper.swift

```diff
@@ -0,0 +1,134 @@
+            let serializer = DataResponseSerializer()
+            let serializerResult = Result { try serializer.serialize(response: response) }
+            let completionResponse = Response(response: response, result: serializerResult)
+            completionHandler(completionResponse)
+        }
+    }
+
+    @discardableResult
+    func request<Value: Decodable>(
+        _ request: SPTDataLoaderRequest,
+        decoder: ResponseDecoder = JSONDecoder(),
+        completionHandler: @escaping (Response<Value, Error>) -> Void
+    ) -> SPTDataLoaderCancellationToken? {
+        return perform(request) { response in
+            let serializer = DecodableResponseSerializer<Value>(decoder: decoder)
+            let serializerResult = Result { try serializer.serialize(response: response) }
+            let completionResponse = Response(response: response, result: serializerResult)
+            completionHandler(completionResponse)
+        }
+    }
+
+    @discardableResult
+    func request(
+        _ request: SPTDataLoaderRequest,
+        options: JSONSerialization.ReadingOptions = [],
+        completionHandler: @escaping (Response<Any, Error>) -> Void
+    ) -> SPTDataLoaderCancellationToken? {
+        return perform(request) { response in
+            let serializer = JSONResponseSerializer(options: options)
+            let serializerResult = Result { try serializer.serialize(response: response) }
+            let completionResponse = Response(response: response, result: serializerResult)
+            completionHandler(completionResponse)
+        }
+    }
+
+    @discardableResult
+    func request<Serializer: ResponseSerializer>(
+        _ request: SPTDataLoaderRequest,
+        serializer: Serializer,
+        completionHandler: @escaping (Response<Serializer.Output, Error>) -> Void
+    ) -> SPTDataLoaderCancellationToken? {
+        return perform(request) { response in
+            let serializerResult = Result { try serializer.serialize(response: response) }
+            let completionResponse = Response(response: response, result: serializerResult)
+            completionHandler(completionResponse)
+        }
+    }
+}
+
+// MARK: - SPTDataLoaderDelegate
+
+extension DataLoaderWrapper: SPTDataLoaderDelegate {
+    func dataLoader(_ dataLoader: SPTDataLoader, didReceiveSuccessfulResponse response: SPTDataLoaderResponse) {
+        if let handler = responseHandlers.removeValue(forKey: response.request.uniqueIdentifier) {
+            handler(response)
+        }
+    }
+
+    public func dataLoader(_ dataLoader: SPTDataLoader, didReceiveErrorResponse response: SPTDataLoaderResponse) {
+        if let handler = responseHandlers.removeValue(forKey: response.request.uniqueIdentifier) {
+            handler(response)
+        }
+    }
+
+    public func dataLoader(_ dataLoader: SPTDataLoader, didCancel request: SPTDataLoaderRequest) {
+        responseHandlers.removeValue(forKey: request.uniqueIdentifier)
```

Comment:
Is there no need for the handling closure to be notified of when a request is cancelled?

Reply:
The need is likely out there, but for the sake of simplicity — while still handling the vast majority of use cases I've seen in our apps — it's been omitted for now.

Reply:
Sounds reasonable!